File: Competition/2019/Blue/Consecutive-Numbers/solution.py


def main():
    
    nums = get_n_process_STDIN()

    consec_nums = 0
    prev_num = None

    for num in nums:

        if num == prev_num:
            consec_nums += 1
        
        else:
            consec_nums = 0
        
        if consec_nums == 2:
            break

        prev_num = num
    
    if consec_nums == 2:
        print(f"Consecutive values found for {prev_num}")
    
    else:
        print("Consecutive values not found")


def get_n_process_STDIN():
    """Get and process STDIN data."""

    print("Enter your numbers separated by a space")
    numbers = input().split()
    numbers = [int(num) for num in numbers]

    return numbers

main()

File: Competition/2019/Blue/Consecutive-Numbers/Solution.java
import java.util.Scanner;
import java.util.ArrayList;

public class Solution
{
    public static void main(String[] args)
    {
        Scanner myScanner = new Scanner(System.in);

        System.out.println("Enter your numbers separated by a space");
        String numsStrForm = myScanner.nextLine();
        myScanner.close();

        String[] numsStrSplit = numsStrForm.split(" ");
        ArrayList<Integer> nums = new ArrayList<Integer>();

        for (String numStr : numsStrSplit)
        {
            int num = Integer.valueOf(numStr);
            
            nums.add(num);
        }

        int consecNums = 0;
        int prevNum = Integer.MIN_VALUE;

        for (int num : nums)
        {
            if (num == prevNum)
            {
                consecNums++;
            }

            else
            {
                consecNums = 0;
            }

            if (consecNums == 2)
            {
                break;
            }

            prevNum = num;
        }

        if (consecNums == 2)
        {
            System.out.println("Consecutive values found for " + prevNum);
        }

        else
        {
            System.out.println("Consecutive values not found");
        }
    }
}